codex-rs/exec/src/event_processor.rs

```diff
@@ -1,15 +1,19 @@
+use std::path::Path;
+
 use codex_common::summarize_sandbox_policy;
 use codex_core::WireApi;
 use codex_core::config::Config;
 use codex_core::model_supports_reasoning_summaries;
 use codex_core::protocol::Event;
 
+use crate::event_processor_with_human_output::CodexStatus;
+
 pub(crate) trait EventProcessor {
     /// Print summary of effective configuration and user prompt.
     fn print_config_summary(&mut self, config: &Config, prompt: &str);
 
     /// Handle a single event emitted by the agent.
-    fn process_event(&mut self, event: Event);
+    fn process_event(&mut self, event: Event, last_message_file: Option<&Path>) -> CodexStatus;
```

Comment:
`last_message_file` feels like something that should be passed to the constructor rather than this method?

Also, it looks like the logic that uses it is similar across both implementations? A trait can have a default implementation for a method as a way to share code, FYI.